### PR TITLE
Parens hugging for Js.t obj types

### LINKED ISSUE #2073: Parens hugging for js.t obj types
```reason
type a = array(
  {
    .
    "name": string,
  },
);
```

Would be nice to refmt as:

```reason
type a = array({
  .
  "name": string,
});
```

### PR DESCRIPTION
**Pull request by anmonteiro**
fixes #2073

### CODE REVIEW

**Comment by IwanKaramazow** [comment]
This is the correct approach.
There's another issue where we should format the `Ptyp_constr` on the same line as `type foo =`
```reason
type foo = array({    -> notice how "array" sticks to "type foo"
...
});
```
This is a much bigger project on it's own, touching the internals of type declaration.
I propose that we solve this in another PR.

File: src/reason-parser/reason_pprint_ast.ml [hunk 1]
```diff
@@ -3037,6 +3037,10 @@ let printer = object(self:'self)
             | [{ptyp_desc = Ptyp_object (l, o) }] when not (isJsDotTLongIdent li.txt) ->
                 label (self#longident_loc li)
                   (self#unparseObject ~wrap:("(",")") l o)
+            | [{ptyp_desc = Ptyp_constr(lii, [{ ptyp_desc = Ptyp_object (ll, o) }])}]
+              when isJsDotTLongIdent lii.txt && List.length ll > 0 ->
```
Comment (chenglou) [inline src/reason-parser/reason_pprint_ast.ml]: Try `ll <> []`. `O(1)` check

Comment (chenglou) [reply src/reason-parser/reason_pprint_ast.ml]: Eh ops, just do `ptyp_desc = Ptyp_object (_::_, o)`

Comment (chenglou) [reply src/reason-parser/reason_pprint_ast.ml]: I'll correct that. Thanks!

**Pull request merged**